### PR TITLE
Feature/upgrade fedstack webpack v3

### PR DESCRIPTION
**Pull request by jwfwessels**
- Updated all packages that I could. 
- Tested on Kopano-web.

### Before: 
Version: webpack 1.13.3
Time: 20780ms
                                             Asset       Size  Chunks             Chunk Names
       kopano-website.main.c7a0c4c37c0a3b296f21.js    1.23 MB       0  [emitted]  main
     kopano-website.styles.275db77bbc8a1e7737a0.js  370 bytes       1  [emitted]  styles
    kopano-website.styles.275db77bbc8a1e7737a0.css    24.3 kB       1  [emitted]  styles
   kopano-website.main.c7a0c4c37c0a3b296f21.js.map    5.67 MB       0  [emitted]  main
 kopano-website.styles.275db77bbc8a1e7737a0.js.map    3.77 kB       1  [emitted]  styles
kopano-website.styles.275db77bbc8a1e7737a0.css.map    75.5 kB       1  [emitted]  styles
### After:
Version: webpack 3.1.0
Time: 18033ms
                                             Asset       Size  Chunks                    Chunk Names
       kopano-website.main.22c85222a5f43637c249.js    1.22 MB       0  [emitted]  [big]  main
     kopano-website.styles.a4d71835786e9045849c.js  564 bytes       1  [emitted]         styles
    kopano-website.styles.a4d71835786e9045849c.css      26 kB       1  [emitted]         styles
kopano-website.styles.a4d71835786e9045849c.css.map    75.2 kB       1  [emitted]         styles

^ seems i still need to optimise the setup a bit.

## Changelog:
- node v7.1.0 => v8.1.3
- most npm packages updated
- webpack v1.x => v3.1.0
- changed some eslint `error`'s to `warnings` cause they were driving ppl crazy.

### CODE REVIEW

**Review by CSergienko — APPROVED** [review]
+1

File: .eslintrc [hunk 1]
```diff
@@ -34,22 +34,22 @@
...
-        "block-spacing": ["error", "always"],
+        "block-spacing": ["warn", "always"],
         "comma-style": ["error", "last"],
-        "comma-spacing": ["error", { "before": false, "after": true }],
-        "indent": ["error", 4, {"SwitchCase": 1}],
+        "comma-spacing": ["warn", { "before": false, "after": true }],
+        "indent": ["warn", 4, {"SwitchCase": 1}],
         "no-ternary": [0],
-        "padded-blocks": ["error", "never"],
+        "padded-blocks": ["warn", "never"],
```
Comment (pringelmann) [inline .eslintrc]: :raised_hands:

**Review by pringelmann — APPROVED** [review]

**Comment by pringelmann** [comment]
@CSergienko @jwfwessels can we re-look CSS ordering? Perhaps flexible grouped ordering?

File: .eslintrc [hunk 2]
```diff
@@ -33,24 +37,36 @@
...
+    "off" or 0 - turn the rule off
+    "warn" or 1 - turn the rule on as a warning (doesn’t affect exit code)
+    "error" or 2 - turn the rule on as an error (exit code is 1 when triggered)
+    */
+
     "rules": {
-        "quotes": ["error", "single", { "avoidEscape": true }],
-        "strict": ["error", "global"],
+        "quotes": [1, "single"],
+    //  "strict": ["error", "global"],
```
Comment (CSergienko) [inline .eslintrc]: Why?

File: .eslintrc [hunk 3]
```diff
@@ -33,24 +37,36 @@
...
         "require-jsdoc": [0],
         "valid-jsdoc": [0],
-        "brace-style": ["error", "1tbs", { "allowSingleLine": false }],
-        "camelcase": ["error", { "properties": "always" }],
-        "block-spacing": ["error", "always"],
-        "comma-style": ["error", "last"],
-        "comma-spacing": ["error", { "before": false, "after": true }],
-        "indent": ["error", 4, {"SwitchCase": 1}],
+        "brace-style": [2, "1tbs", { "allowSingleLine": false }],
+    //  "camelcase": ["error", { "properties": "always" }],
```
Comment (CSergienko) [inline .eslintrc]: Why?

File: package.json [hunk 4]
```diff
@@ -6,40 +6,47 @@
   "scripts": {
     "build": "NODE_ENV=production webpack --colors --progress",
     "build:dev": "NODE_ENV=development webpack-dashboard -- webpack-dev-server"
```
Comment (CSergienko) [inline package.json]: Missing comma on end of line.

Comment (CSergienko) [inline package.json]: Missing comma on end of line.

File: webpack.config.js [hunk 5]
```diff
@@ -56,67 +55,63 @@ const BASE_CONFIG = {
...
 
 // Stitch together the correct config based on the environment.
-function configBuilder(process, config) {
+function configBuilder(process, config, env) {
     let mergedConfig;
 
     switch (process) {
+        // #################
+        // Profile Build
+        // TODO: refactor config to not tie build script name to exicution type any more.
```
Comment (CSergienko) [inline webpack.config.js:89]: Please elaborate on this?

Comment (jwfwessels) [reply webpack.config.js:89]: the stuff i just explained to you...

**Pull request merged**